kernel/thread.c

```diff
@@ -100,9 +100,9 @@ void _impl_k_busy_wait(u32_t usec_to_wait)
 #if !defined(CONFIG_ARCH_HAS_CUSTOM_BUSY_WAIT)
 	/* use 64-bit math to prevent overflow when multiplying */
 	u32_t cycles_to_wait = (u32_t)(
-		(u64_t)usec_to_wait *
-		(u64_t)sys_clock_hw_cycles_per_sec() /
-		(u64_t)USEC_PER_SEC
+		usec_to_wait *
+		sys_clock_hw_cycles_per_sec() /
+		USEC_PER_SEC
```

Comment:
This looks dangerous too.  Presumably those casts were there to guarantee that the math was done in 64 bit precision.  So now that that multiply is going to be done in 32 bits and potentially overflow where it didn't before.

Reply:
Initially I was changing I was changing the constant to be long long. Then other problems appeared, I've undone the changes and forgot it.
The MISRA-C violation is promote a 32 wide bit variable to a 64.